zap/src/main/java/org/zaproxy/zap/extension/script/ExtensionScript.java

```diff
@@ -1039,6 +1046,9 @@ private int addScriptsFromDir(File dir, ScriptType type, String targetEngineName
                                             type,
                                             false,
                                             f);
+                            if (enableScripts) {
+                                this.setEnabled(sw, true);
+                            }
```

Comment:
This is enabling before actually loading the contents of the script, for `extender` scripts it means they will not be properly installed (i.e. errors or no-op).

Scripts which are in the directories but whose engine is installed later are not enabled, I think they should but if the behaviour is documented I'm fine either way.

Reply:
Ping.

Reply:
I got carried away with some other work, I'll try to get back to this in a week's time, sorry for keeping this open.

Reply:
No worries and thank you!

Reply:
For the record, the first part of the comment is no longer a problem (at least with the engines provided by add-ons). The second part of the comment will be addressed with docs.